src/CalcManager/CEngine/scicomm.cpp

```diff
@@ -870,153 +863,91 @@ void CCalcEngine::DisplayAnnounceBinaryOperator()
 // we have this separate table to get its localized name and for its Inv function if it exists.
 typedef struct
 {
-    int idsFunc;    // index of string for the unary op function. Can be NULL, in which case it same as button name
-    int idsFuncInv; // index of string for Inv of unary op. Can be NULL, in case it is same as idsFunc
-    bool fDontUseInExpEval;  // true if this cant be used in reverse direction as well, ie. during expression evaluation
+    wstring degreeString; // Used by default if there are no rad or grad specific strings.
+    wstring inverseDegreeString; // Will fall back to degreeString if empty
+
+    wstring radString;
+    wstring inverseRadString; // Will fall back to radString if empty
+
+    wstring gradString;
+    wstring inverseGradString; // Will fall back to gradString if empty
+
+    bool hasAngleStrings = ((!radString.empty()) || (!inverseRadString.empty()) || (!gradString.empty()) || (!inverseGradString.empty()));
```

Comment:
If just one of these strings is not empty, it's sufficient to say the struct `hasAngleStrings`?  Do `degreeString` and `inverseDegreeString` need to be checked here as well?

Reply:
Yes it is sufficient for the purpose we are using it for degree strings act as the default, we will only look for rad/grad versions if hasAngleStrings is true.